tasties_app/views.py

```diff
@@ -7,3 +10,13 @@ def index(request):
 
 def base(request):
     return render(request, 'tasties_app/base.html',)
+
+
+def recipes(request):
+    recipes_list = Recipe.objects.all()
+    recipes_with_ratings = {}
+    for recipe in recipes_list:
+        recipes_with_ratings[recipe] = Rating.objects.filter(recipe_id=recipe).aggregate(Avg('rating'))['rating__avg']
+    recipes_with_ratings = OrderedDict(sorted(recipes_with_ratings.items(), key=lambda x: x[1], reverse=True))
```

Comment:
Maybe there is a more readable way to sort this? Something like you can do with QuerySet `Member.objects.all().order_by('firstname').values()`

Reply:
@natifridman Please notice that `Rating` is not a FK in `Recipe`. Therefore, when we want the rating average for a recipe, we need to extract that value from a QuerySet of `Rating`s with the field `recipe_id` matching the desired `Recipe`.
I personally can't think of a method that would be more readable than what Gil already did.

Reply:
I agree with Efraim.
If I had a `rating` field in `Recipe` I could do as you suggested.
But instead I'm using a dictionary here to calculate each recipe's rating and assign them together.